cmd/serve/httplib/serve/dir.go

```diff
@@ -80,9 +83,14 @@ func (d *Directory) Serve(w http.ResponseWriter, r *http.Request) {
 
 	fs.Infof(d.DirRemote, "%s: Serving directory", r.RemoteAddr)
 
-	err := d.HTMLTemplate.Execute(w, d)
+	buf := &bytes.Buffer{}
+	err := d.HTMLTemplate.Execute(buf, d)
 	if err != nil {
 		Error(d.DirRemote, w, "Failed to render template", err)
 		return
 	}
+	_, err = buf.WriteTo(w)
+	if err != nil {
+		Error(d.DirRemote, nil, "Failed to drain template buffer", err)
```

Comment:
This doesn't solve the original problem. As soon as the first `w.Write` attempt has been made, `w.WriteHeader` was automatically called so `serve.Error` shouldn't be called in this case.

Reply:
serve.Error wont write to the http response tho, instead its only in the log

Reply:
Maybe update the doc then? :) `// Error returns an http.StatusInternalServerError and logs the error`

Reply:
Done :smile: